internal/venice-common/src/main/java/com/linkedin/venice/system/store/ControllerClientBackedSystemSchemaInitializer.java

```diff
@@ -142,19 +145,31 @@ public void execute() {
     for (int version = 1; version <= protocolDefinition.getCurrentProtocolVersion(); version++) {
       Schema schemaInLocalResources = schemasInLocalResources.get(version);
       if (schemaInLocalResources == null) {
-        throw new VeniceException(
-            "Invalid protocol definition: " + protocolDefinition.name() + " does not have a version " + version
-                + " even though it is less than or equal to the current version ("
-                + protocolDefinition.getCurrentProtocolVersion() + ").");
-      }
-      checkAndMayRegisterValueSchema(storeName, version, schemasInZk.get(version), schemaInLocalResources);
+        if (inputSchemas == null) {
```

Comment:
Curious, why do we need this if branch? From above logic:
```
Map<Integer, Schema> schemasInLocalResources =
        inputSchemas != null ? inputSchemas : Utils.getAllSchemasFromResources(protocolDefinition);
```
If `schemaInLocalResources` is null, `inputSchemas` must be null, right?

Reply:
I know it's a bit confusing, but `schemaInLocalResources` is defined one line above (not the same as `schemasInLocalResources`).

So, in its usage, e.g. `schemaInitializer.execute(ImmutableMap.of(schemaId, schema));`, the passed in map contains only 1 entry, so we could have schemaInLocalResources == null for some versions, which is a valid case, and we not throw exception in this case.

Reply:
I see. I missed `s` in the variable name. Thanks for the explanation. It brings up another question: if a controller or server finds a new unknown KME schema in the message header, the version of the new schema will be larger than `protocolDefinition.getCurrentProtocolVersion()`. In this case, `schemaInLocalResources` will be null and the code here won't register the new KME schema. Please correct me if I missed something.